ITISFoundation/osparc-simcore
# ✨ New osparc variable `OSPARC_VARIABLE_API_HOST` (⚠️)

## What do these changes do?


Exposes new osparc variable `OSPARC_VARIABLE_API_HOST` which corresponds to the base url to the public api e.g. `https://api.osparc.io` 



Next step:
- Default osparc-api clients configuration to read these the env vars above


## Related issue/s

- Resolves ITISFoundation/osparc-issues#1389


## How to test

- Publish a new service (e.g. jupyter notebook) with these environments (NOTE: $$ to avoid substitution when injected in the labels)
```yaml
      environment:                                                                                                                                  
        - OSPARC_API_HOST=$${OSPARC_VARIABLE_API_HOST}                                                                                                
        - OSPARC_API_KEY=$${OSPARC_VARIABLE_API_KEY}                                                                                                
        - OSPARC_API_SECRET=$${OSPARC_VARIABLE_API_SECRET}
```
- Start service and `printenv` should contain correct values to access the api
- `pip install osparc` and run
```python
from osparc import ApiClient, UsersApi, Confirmation
import os

cfg = Configuration(
    host=os.environ["OSPARC_API_HOST"],
    username=os.environ["OSPARC_API_KEY"],
    password=os.environ["OSPARC_API_SECRET"],
)
print(cfg.host)

with ApiClient(cfg) as api_client:
    users_api = UsersApi(api_client)
    profile = users_api.get_my_profile()
    print(profile)
```

## Dev-ops checklist ⚠️

- [x] https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/554
